File: que3.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
#include <map>
using namespace std;


int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */   
    int t;
    cin>>t; //number of testcases
    for(int i=0;i<t;i++){
        int n;
        cin>>n;//number of trees
        // we are creating a list with all the fruits
        vector<int>fruits;
        for(int j=0;j<n;j++){
            int f;
            cin>>f;
            fruits.emplace_back(f);
        }
        // we will create a dictionary==window, with number of fruits of each type
        int left = 0;
        map<int,int>count;
        int best = 0;
        for(int right=0;right<n;right++){
            int f = fruits[right];
            count[f]++;
            // our window can hold only two type of fruits , so we will shrink it if it exceeds that value
            while(count.size()>2){
                int left_f = fruits[left];
                count[left_f]--;
                if(count[left_f]==0){
                    count.erase(left_f);
                }
                left++;
            }
            // total number of fruits
            best = max(best,right-left+1);
        }
        cout<<best<<endl; // printing the final output
        
    }
    return 0;
}

File: que3.py
t = int(input()) # input for number of testcases

for _ in range(t): 
    n = int(input()) # input for number of trees
    fruits = list(map(int,input().split())) # list containing all types of fruits
    # we will consider a window
    left = 0 # starting point of the window
    count = {} # how many fruits of eacch type are inside the window
    best = 0 # biggest window found so far
    
    for right in range(n): # the right loop is for accessing all the elements of the fruits list
        f = fruits[right]
        count[f] = count.get(f,0) +1#dictionary which tells how many fruits of each type are there
        # if we have 3 types we need to shrink the window as the basket can hold only two types
        while len(count)>2:
            left_f = fruits[left]
            count[left_f] -=1
            if count[left_f] ==0:
                del count[left_f]
            left +=1
            
        best = max(best,right-left+1)
        
    print(best)